### PR TITLE
Fix default badge bug

### PR DESCRIPTION
**Pull request by joyceqin-stripe**
## Summary

Replace row button for the previous default payment method if it wasn't also previously selected
## Motivation

Bug bash bug
Default badge showed up on more than one row button because we were only replacing the previously selected row button, but the previously selected row button is not necessarily the default, so there was a scenario where the default badge did not get cleared from the previous default.

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/VerticalSavedPaymentMethodsViewController.swift [hunk 1]
```diff
@@ -399,12 +399,17 @@ extension VerticalSavedPaymentMethodsViewController: UpdatePaymentMethodViewCont
     }
 
     private func updateDefault(paymentMethod: STPPaymentMethod) async throws {
+        let previousDefaultPaymentMethod = defaultPaymentMethod
         _ = try await savedPaymentMethodManager.setAsDefaultPaymentMethod(defaultPaymentMethodId: paymentMethod.stripeId)
         defaultPaymentMethod = paymentMethod
         // if there was a previously selected payment method, replace it to deselect it and remove the badge if it was default
         if let previousSelectedPaymentMethod {
             replace(paymentMethod: previousSelectedPaymentMethod, with: previousSelectedPaymentMethod, selectedState: .unselected)
         }
```
[the remark this answers is not in the document]
Comment (joyceqin-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/VerticalSavedPaymentMethodsViewController.swift:408]: When we set a new default payment method, we select it (at least, according to the Figma), so we need to deselect the previously selected payment method

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/VerticalSavedPaymentMethodsViewController.swift [hunk 2]
```diff
@@ -399,12 +399,17 @@ extension VerticalSavedPaymentMethodsViewController: UpdatePaymentMethodViewCont
...
+        // if there was a previous default payment method that wasn't previously selected, replace it to remove the badge
+        if let previousDefaultPaymentMethod, previousDefaultPaymentMethod != previousSelectedPaymentMethod {
```
[the remark this answers is not in the document]
Comment (joyceqin-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/VerticalSavedPaymentMethodsViewController.swift:410]: We replace/remove the default badge no matter what. We have up to 3 row buttons to replace: the previously selected payment method (to deselect it), the previous default payment method (to remove the badge), and the new default payment method (to add the badge and select it).

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift [hunk 3]
```diff
@@ -2822,6 +2822,47 @@ class PaymentSheetDefaultSPMUITests: PaymentSheetUITestCase {
         // Ensure checkbox is not displayed if it's already the default
         XCTAssertFalse(app.switches["Set as default payment method"].waitForExistence(timeout: 3))
     }
+
+    func testSetAsDefaultVerticalNoDuplicateBadges_CustomerSession() {
```
[the remark this answers is not in the document]
Comment (joyceqin-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:2826]: I'm not sure how useful a test there would be. I could just remove the test though.

[the remark this answers is not in the document]
Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:2826]: If a test in there isn't useful, why is this test useful?

[the remark this answers is not in the document]
Comment (joyceqin-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:2826]: This test would fail on master, but not this branch. I can remove the test since it's pretty niche and unlikely to be an issue going forward, but this test specifically required setting a default, selecting a payment method that isn't the default, and then setting a new default. The reason multiple default badges would show up previously was that it made the incorrect assumption that replacing the previously selected payment method would also remove the badge, but the previously selected payment method is not guaranteed to be the previous default, so it would not remove the badge. I'm not really sure how to test that in VerticalSavedPaymentMethodsViewControllerTests.

[the remark this answers is not in the document]
Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:2826]: Yeah can see how unit testing could be difficult since we need an ephemeral key, given the edge case this is, I'm. peaceful removing this test.
